main.cpp

```diff
@@ -107,6 +128,76 @@ Thing 4) analog console
     3) add tone
  */
 
+/*
+Thing 5) Mixer
+5 properties:
+    1) faders 
+    2) routing panel 
+    3) record ready btn
+    4) input fx
+    5) input selector
+3 things it can do:
+    1) record audio on track
+    2) pan sound
+    3) change audio volume  
+ */
+
+/*
+Thing 6) Virtual MIDI Keyboard
+5 properties:
+    1) number of octaves 
+    2) number of MIDI channel
+    3) keyboard size in pixel 
+    4) key colours
+    5) velocity range 
+3 things it can do:
+    1) send midi notes 
+    2) display MIDI numbers (note to MIDI)
+    3) recieve input from computer keyboard
+ */
+
+/*
+Thing 7) Toolbar Docker
+5 properties:
+    1) number of tools in docker
+    2) tool selector size  
+    3) docker position 
+    4) docker size
+    5) hide docker btn 
```

Comment:
5): This is an object, not a primitive!   You need to pick things that can be represented with `int float double bool char std::string`
- [ ] revise